pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -284,17 +284,18 @@ public class ModelInterpreter implements Serializable {
      * Executes the given closure inside a declarative agent block, if appropriate.
      *
      * @param context Either a stage or root object, the context we're running in.
+     * @param root The root object for this pipeline
      * @param agent The agent context we're running in
      * @param body The closure to execute
      * @return The return of the resulting executed closure
      */
-    def inDeclarativeAgent(Object context, Agent agent, Closure body) {
+    def inDeclarativeAgent(Object context, Root root, Agent agent, Closure body) {
```

Comment:
Agreed. Opened [JENKINS-42114](https://issues.jenkins-ci.org/browse/JENKINS-42114) for that.